cmflib/cmf.py

```diff
@@ -372,10 +396,10 @@ def create_execution(
         # Assigning current file name as stage and execution name
         current_script = sys.argv[0]
         file_name = os.path.basename(current_script)
-        name_without_extension = os.path.splitext(file_name)[0]
+        self.name_without_extension = os.path.splitext(file_name)[0]    # Assigning to an instance variable

```

Comment:
Why cant we leave this as local variable ? Making this instance variable seems unnecessary . Check other ways we can avoid the errors in static type checking